specification/application/conventions.tex

```diff
@@ -7,14 +7,34 @@ \section{Application-level conventions}\label{sec:application_level_conventions}
 
 \subsection{Node identifier distribution}
 
-An overview of related concepts is provided in chapter \ref{sec:basic_concepts}.
+An overview of related concepts is provided in chapter~\ref{sec:basic}.
 
 Valid values of node-ID range from 0 up to a transport-specific upper boundary
 which is guaranteed to be above 127 for any transport.
 
 The two uppermost node-ID values are reserved for diagnostic and debugging tools;
 these node-ID values should not be used in fielded systems.
 
+\subsection{Timing constraints}
+
+\subsubsection{Recommended message publication timing constraints}
+
+Generally, it is recommended to abort a message transmission if it cannot be completed in 1 second.
+
+It is expected that high-frequency high-priority messages may opt for lower timeout values,
+whereas low-priority delayable data may opt for higher timeout values to account for network congestion.
+Refer to section~\ref{sec:transport_transfer_priority} on transfer prioritization.
+
+\subsubsection{Recommended service invocation timing constraints}
+
+It is recommended to abort a service transfer transmission if it cannot be completed in 1 second.
+It is recommended to stop waiting for a service response if it could not be received in the same amount of time;
+upon reaching this condition, the service request operation should be considered unsuccessful.
+
+If the server uses a significant part of the timeout period to process the request,
+the client might drop the request before receiving the response.
+It is recommended to ensure that the server will be able to process any request in less than 0.5 seconds.
```

Comment:
I think the service request timing recommendations do belong to the application layer since they are directly related to the application's behavior, so I don't think that this recommendation is misplaced.

It might be, however, overconstraining, I agree. Maybe we should reduce this to generic advice that service implementations should be mindful of their latency?

Reply:
While mindfulness is laudable we should always provide specific effects. Perhaps what this needs is a link to the transfer ID with a discussion on how increased latency can lead to invalid data. We should also provide a formula to allow system integrators to calculate the allowable latency before the transfer ID becomes invalid with variable bit-width transfer IDs considered (for future transports). I think we mention 2-seconds somewhere so we have a bunch of time constants that aren't strongly linked in the specification but which control the correctness of multi-part transfers.